src/main/java/software/amazon/documentdb/jdbc/DocumentDbDatabaseMetaData.java

```diff
@@ -498,9 +498,9 @@ public ResultSet getPrimaryKeys(final String catalog, final String schema,
         final List<List<Object>> metaData = new ArrayList<>();
         final String regexSchemaPattern = convertPatternToRegex(schema);
         final String regexTableSchemaPattern = convertPatternToRegex(table);
-        if (schema == null || properties.getDatabase().matches(regexSchemaPattern)) {
+        if (isNullOrWhitespace(schema) || properties.getDatabase().matches(regexSchemaPattern)) {
```

Comment:
I might be missing context here but as I understand it, null and whitespace don't mean the same thing in getPrimaryKeys/getImportedKeys, `""` means without a schema whereas `null` means don't use the schema to narrow search. Not sure if I've understood this correctly but it seems that these should do slightly different things.  [reference](https://docs.oracle.com/javase/7/docs/api/java/sql/DatabaseMetaData.html#getImportedKeys(java.lang.String,%20java.lang.String,%20java.lang.String))

Reply:
yea `""` and null do mean different things. The reason I made the change is that ODBC can only pass `""` to the JNI call and cannot pass `null` to my understanding, so I'm just enabling JDBC to send non-empty result sets. So basically when we pass `""` from ODBC we are trying to mean the same thing as `null`. I think since we implemented `getColumns` and `getTables` by treating `""` the same as `null` for schema and catalog, it should be okay?

Reply:
That seems reasonable

Reply:
this might break the behaviour of importedkeys tho, maybe you will need to have other API just for ODBC

Reply:
just because we need the different results and differentiate when is null or is empty
https://docs.microsoft.com/en-us/sql/odbc/reference/syntax/sqlprimarykeys-function?view=sql-server-ver15

Reply:
I have instead reverted my code changes for getPrimaryKeys and getImportedKeys. I have updated the tests to make sure the behavior from JDBC is correct when empty strings are passed to getPrimaryKeys and getImportedKeys.